lnpbp-0021.md

```diff
@@ -1,213 +1,182 @@
-            // salt value from state transition metadata.
-            // NB: It is unique inside single state transition only, not
-            // globally. For global unique id use non-convoluted hash value.
-            format: StateFormat::CustomData(DataFormat::Unsigned(Bits::Bit32, 0, core::u32::MAX as u128)),
-            abi: bmap! {
-                // Here we ensure that each unique state value is
-                // transferred once and only once (using "salt" value for
-                // collision resoultion)
-                AssignmentAction::Validate => Procedure::Embedded(StandardProcedure::IdentityTransfer)
-            }
-        },
-        OwnedRightsType::Renomination => StateSchema {
-            format: StateFormat::Declarative,
-            abi: none!()
-        }
-    },
-    public_right_types: none!(),
-}
+- [Abstract](#abstract)
+- [Background](#background)
+- [Motivation](#motivation)
+- [Design](#design)
+- [Specification](#specification)
+- [Compatibility](#compatibility)
+- [Rationale](#rationale)
+- [Reference implementation](#reference-implementation)
+- [Acknowledgements](#acknowledgements)
+- [References](#references)
+- [Copyright](#copyright)
+- [Test vectors](#test-vectors)
+
+
+## Abstract
+
+
+## Background
+
+
+## Motivation
+
+
+## Design
+
+- [x] Media as URI (which can be attachments distributed with Storm or usual URLs)
+- [x] Small media ("previews" if given together with large media)
+- [x] Fraction of assets
+- [x] Engravings
+- [x] Reserves
+- [x] Possible decentralized issue
+
+
+## Specification
+
+Interface specification is the following Contractum code:
+
+```haskell
+-- # Defining main data structures
+
+-- collectibles are usually scarse, so we limit their max number to 64k
+data ItemsCount :: U16
+
+-- each collectible item is unique and must have an id
+data TokenId :: U16
+
+-- Collectibles are owned in fractions of a single item; here the owned
+-- fraction means `1/F`. Ownership of the full item is specified `F=1`;
+-- half of an item as `F=2` etc.
+data OwnedFraction :: U64
+
+data TxOut :: txid [Byte ^ 32], vout U16
+
+-- allocation of a single token or its fraction to some transaction output
+data Allocation :: TokenId, OwnedFraction
+
+data Media ::
+    type MimeType,
+    data [Byte]
+
+data Attachment ::
+    type MimeType,
+    uri: Uri
+
+data POR :: -- proof of reserves
+    reserves TxOut,
+    proof [Byte] -- schema-specific proof 
+
+data Token ::
+    id TokenId,
+    name [Ascii ^ 1..40],
+    details [Unicode ^ 40..256]?,
+    preview Media?, -- always embedded preview media < 64kb
+    media Attachment?, -- external media which is the main media for the token
+    attachments { U8 ^ ..20 -> Attachment } -- auxillary attachments by type (up to 20 attachments)
+    reserves POR? -- output containing locked bitcoins; how reserves are
+                  -- proved is a matter of a specific schema implementation
+
+data Nomination :: 
```

Comment:
```suggestion
data Nominal :: 
```

Reply:
One thing, I think we should use the term Denomination here, as a reference to this:
https://en.wikipedia.org/wiki/Denomination_(currency)#Choice_of_name

The definition for the word Nominal has some adjacent meanings, but isn't as applicable. See:
https://www.merriam-webster.com/dictionary/nominal

I can have an English scholar provide his opinion if that helps.

Reply:
As an alternative I think also something like "Identifier(s)" could fit.

Reply:
@fedsten Not sure if you saw the email thread I included you two on, but Prof Andrew M Bailey suggested the same thing :)

Reply:
I think `Identifier` is too general and used in many other places, so leaving `Denomination`